File: app/src/main/java/com/fentury/testapp/utils/StringFormatter.java
package com.fentury.testapp.utils;

import android.content.Context;
import android.content.res.Resources;

/**
 * Created by Vadim on 20.11.2016.
 */

public class StringFormatter {
    Context context;

    public StringFormatter(Context context) {
        this.context = context;
    }

    public String formatString(int resId, String str) {
        String result1 = "";
        try {
            result1 = context.getString(resId);
        } catch (Resources.NotFoundException e) {

        }
        return String.format("%s %s", result1, (str == null) ? "" : str).trim();
    }
}

File: app/src/main/java/com/fentury/testapp/utils/StringFormatter.kt
package com.fentury.testapp.utils

import android.content.Context
import android.content.res.Resources

/**
 * Created by morozov on 12/5/16.
 */
class StringFormatter(val context: Context) {

    fun formatString(resId: Int, str: String?): String {
        var result1 = ""

        try {
            result1 = context.getString(resId)
        } catch (e: Resources.NotFoundException) {

        }
        return String.format("%s %s", result1, str ?: "" ).trim { it <= ' ' }
    }
}